src/Chapter2.hs

```diff
@@ -621,8 +626,14 @@ Write a function that takes elements of a list only on even positions.
 >>> takeEven [2, 1, 3, 5, 4]
 [2,3,4]
 -}
-takeEven = error "takeEven: Not implemented!"
-
+takeEven :: [Int] -> [Int]
+takeEven [] = []
+takeEven l = go [] l
+  where
+    go :: [Int] -> [Int] -> [Int]
+    go acc [] = acc
+    go acc (x:_:xs) = go (acc ++ [x]) xs
+    go acc (x:_) = go (acc ++ [x]) []
```

Comment:
Usually, adding elements to the end of the list is costly. Your implementation is correct :+1: But Alternatively, you could use pattern matching in `takeEven` instead. As you need to take every second element, it is possible to represent with a few pattern matching clauses 🙂